### ISSUE TITLE
Quarto VS Code extension inadvertently disables GitHub Copilot autocomplete for all other languages

### ISSUE THREAD

**Issue by hangler** [quarto-dev/quarto#621]
Quarto is wonderful. However, the VS Code extension sets the following configuration rule by default:

```
        "github.copilot.enable": {
		"quarto": false
	}
```

When `github.copilot.enable` has a value set for any particular language, though, it implicitly **disables** Copilot for all non-listed languages. This is unexpected and difficult to debug.

The solution is to add a wildcard setting:

```
        "github.copilot.enable": {
		"quarto": false
                "*": true
	}
```

However, I'm not sure how to have this interact smoothly with any settings a user might already have for this configuration rule. (In my case, nothing was set before installing Quarto.)

**Comment by cscheid** [comment]
Sorry, I just want to make sure I understand. You're claiming that this setting disabled copilot for you in _all_ languages? I have the extension installed and can't repro (I use copilot on a daily basis):

https://github.com/user-attachments/assets/a1617903-c580-434f-a8e7-4b32b0885378

**Comment by hangler** [comment]
Yes, in my case it did. Here's an example with a [Remix](https://remix.run/docs/en/main/start/quickstart) project I just created:

https://github.com/user-attachments/assets/60134afd-1763-4a35-ad7f-db252b19ad28

Details on my VS Code:

```
Version: 1.95.3 (Universal)
Commit: f1a4fb101478ce6ec82fe9627c43efbf9e98c813
Date: 2024-11-13T14:50:04.152Z
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Darwin arm64 24.1.0
```

Happy to provide further details on my setup as required!

**Comment by cscheid** [comment]
That appears to be exactly what I'm running:

```
Version: 1.95.3 (Universal)
Commit: f1a4fb101478ce6ec82fe9627c43efbf9e98c813
Date: 2024-11-13T14:50:04.152Z (4 wks ago)
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Darwin arm64 22.6.0
```

**Comment by hangler** [comment]
Did you possibly have other values set on the `github.copilot.enable` object in your settings before adding the `"quarto": false` line at all..?

**Comment by cscheid** [comment]
I'll say that "When `github.copilot.enable` has a value set for any particular language, though, it implicitly disables Copilot for all non-listed languages" smells like a VS Code bug to me. Nevertheless, it's a sufficiently bad one that we should be willing to work around it.

Adding `"*": true` seems intuitive, but I'd like to know how VS Code composes `github.copilot.enable` settings before making this change. It might just be that we want to instead remove the `github.copilot.enable` entry altogether.

**Comment by hangler** [comment]
> I'll say that "When `github.copilot.enable` has a value set for any particular language, though, it implicitly disables Copilot for all non-listed languages" smells like a VS Code bug to me.

I was just thinking exactly this myself, as I pondered it some more... setting a value specific to one language doesn't seem like it should require opting-in for all other languages! (I just tested, and changing the value to `"quarto": true`, but not adding the wildcard entry, has the same effect as `"quarto": false`).

**Comment by mcanouil** [comment]
FYI, the extension is disabling Copilot for Quarto documents:

- https://github.com/quarto-dev/quarto/blob/1375251f147d7f9417be8f41181c8b4e3b9f7e3a/apps/vscode/package.json#L904-L906
